### PR TITLE
Tests of Dynamic Checks at Array Subscript and Deref

### PR DESCRIPTION
**Pull request by lenary**
These are tests to go with Microsoft/checkedc-clang#163

### CODE REVIEW

**Comment by lenary** [comment]
I am yet to write tests for deref checks, will do that this afternoon, expect a second commit.

**Comment by lenary** [comment]
I've added the tests for local arrays, will get to parameters soon this afternoon. Also I found I could split the -O3 tests into a separate file (which will run the same code), which keeps the test suite quick.

File: tests/dynamic_checking/deref-arith-check-opt.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// Tests that the Dereference Operator works with dynamic checks and pointer arithmetic
+//
+// The following lines are for the clang automated test suite
```
Comment (dtarditi) [inline tests/dynamic_checking/deref-arith-check-opt.c]: Could you add a comment describing what this file does?   It is important to have a comment explaining that the run lines compile another file other than this one.

File: tests/dynamic_checking/deref-check-opt.c [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// Tests that the Dereference Operator works with dynamic checks
+//
+// The following lines are for the clang automated test suite
```
Comment (dtarditi) [inline tests/dynamic_checking/deref-check-opt.c]: This too needs a comment describing what this is doing.

File: tests/dynamic_checking/subscript-check-opt.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Tests that the Array Subscript Operator works with dynamic checks
+//
+// In particular, this re-runs the tests in subscript-check.c with Optimizer at O3
```
Comment (dtarditi) [inline tests/dynamic_checking/subscript-check-opt.c]: This needs to point out that that the RUN line actually compilers another file other than this one.

**Pull request merged**